plugins/module_utils/network/sonic/config/stp/stp.py

```diff
@@ -0,0 +1,1288 @@
+
+    def get_interfaces_list(self, interfaces):
+        intf_list = []
+        for intf in interfaces:
+            intf_dict = {}
+            intf_cfg_dict = {}
+            intf_name = intf.get('intf_name', None)
+            cost = intf.get('cost', None)
+            port_priority = intf.get('port_priority', None)
+
+            if intf_name:
+                intf_cfg_dict['name'] = intf_name
+            if cost:
+                intf_cfg_dict['cost'] = cost
+            if port_priority:
+                intf_cfg_dict['port-priority'] = port_priority
+            if intf_cfg_dict:
+                intf_dict['name'] = intf_name
+                intf_dict['config'] = intf_cfg_dict
+                intf_list.append(intf_dict)
+
+        return intf_list
+
+    def get_vlans_common(self, vlans, cfg_vlans):
+        """Returns the vlan ranges that are common in the want and have
+        vlans lists
+        """
+        vlans = self.get_vlan_id_list(vlans)
+        cfg_vlans = self.get_vlan_id_list(cfg_vlans)
+        return self.get_vlan_range_list(list(set(vlans).intersection(set(cfg_vlans))))
+
+    def get_vlans_diff(self, vlans, cfg_vlans):
+        """Returns the vlan ranges present only in the want vlans list
+        and not in the have vlans list
+        """
+        vlans = self.get_vlan_id_list(vlans)
+        cfg_vlans = self.get_vlan_id_list(cfg_vlans)
+        return self.get_vlan_range_list(list(set(vlans) - set(cfg_vlans)))
+
+    @staticmethod
+    def get_vlan_id_list(vlans):
+        """Returns a list of all VLAN IDs specified in a vlans list"""
+        vlan_id_list = []
+
+        if vlans:
+            for vlan_val in vlans:
+                if '-' in vlan_val:
+                    start, end = vlan_val.split('-')
+                    vlan_id_list.extend(range(int(start), int(end) + 1))
+                else:
+                    # Single VLAN ID
+                    vlan_id_list.append(int(vlan_val))
+
+        return vlan_id_list
+
+    @staticmethod
+    def get_vlan_range_list(vlan_id_list):
+        """Returns the vlans list for a given list of VLAN IDs"""
+        vlan_range_list = []
+
+        if vlan_id_list:
+            vlan_id_list.sort()
+            for vlan_range in get_ranges_in_list(vlan_id_list):
+                vlan_range_list.append('-'.join(map(str, (vlan_range[0], vlan_range[-1])[:len(vlan_range)])))
+
+        return vlan_range_list
+
+    def convert_vlans_list(self, vlans):
+        converted_vlans = []
+
+        for vlan in vlans:
+            if len(vlan) == 1:
+                converted_vlans.append(int(vlan))
+            else:
+                converted_vlans.append(vlan.replace('-', '..'))
+
+        return converted_vlans
+
+    def get_delete_stp_requests(self, commands, have, is_delete_all):
+        requests = []
+
+        if not commands:
+            return requests
+
+        if is_delete_all:
+            requests.append(self.get_delete_all_stp_request())
+        else:
```

Comment:
Are you talking about for overridden? Like we discussed, I updated the state with the wrong variable. I have corrected it.

Reply:
For "overridden" handling this looks fine with the latest changes.

But for "deleted" state, the commands logged via "update_states" should only specify commands for configuration that have been deleted from the switch. For the "delete all" case, the "deleted" state handling is fine. But for other delete cases, there may be deletion commands that did not result in deletion of anything on the switch.

For this case, the "commands" dict should not be the same as the "invocation" dict.

Reply:
It's apparent that commands for delete state hasn't been handled properly in several existing modules. This needs to be addressed. I have handled commands for global deletion and will have to address the other protocols in the next release since it requires more rework.